From 9a4e363bf5739e5e812de5cb9dcb33b75c007943 Mon Sep 17 00:00:00 2001
From: Lyon Wang <lyon.wang@nxp.com>
Date: Wed, 18 Apr 2018 15:04:32 +0800
Subject: [PATCH 09/68] Revert "player: Fix-up set_seek_accurate()
 configuration to take a player config instead of a player instance"

This reverts commit cc58bd6ae071dec4ea7b4be626034accd0372755.

Revert this commits, as in set_config()
Config flag can only be set in STOPPED state

Upstream-Status: Pending
(cherry picked from commit b6922ed098879deeda0e87c9a4b35e37449ef7a8)
Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 gst-libs/gst/play/gstplay.c | 5 +++--
 gst-libs/gst/play/gstplay.h | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/gst-libs/gst/play/gstplay.c b/gst-libs/gst/play/gstplay.c
index a8b65f3..7ca97f2 100644
--- a/gst-libs/gst/play/gstplay.c
+++ b/gst-libs/gst/play/gstplay.c
@@ -4514,7 +4514,7 @@ gst_play_config_get_position_update_interval (const GstStructure * config)
 
 /**
  * gst_play_config_set_seek_accurate:
- * @config: a #GstPlay configuration
+ * @play: #GstPlay instance
  * @accurate: accurate seek or not
  *
  * Enable or disable accurate seeking. When enabled, elements will try harder
@@ -4530,8 +4530,9 @@ gst_play_config_get_position_update_interval (const GstStructure * config)
  * Since: 1.20
  */
 void
-gst_play_config_set_seek_accurate (GstStructure * config, gboolean accurate)
+gst_play_config_set_seek_accurate (GstPlay * self, gboolean accurate)
 {
+  GstStructure *config = self->config;
   g_return_if_fail (config != NULL);
 
   gst_structure_id_set (config,
diff --git a/gst-libs/gst/play/gstplay.h b/gst-libs/gst/play/gstplay.h
index 164376f..2c53aeb 100644
--- a/gst-libs/gst/play/gstplay.h
+++ b/gst-libs/gst/play/gstplay.h
@@ -373,7 +373,7 @@ GST_PLAY_API
 guint          gst_play_config_get_position_update_interval  (const GstStructure * config);
 
 GST_PLAY_API
-void           gst_play_config_set_seek_accurate (GstStructure * config, gboolean accurate);
+void           gst_play_config_set_seek_accurate (GstPlay * play, gboolean accurate);
 
 GST_PLAY_API
 gboolean       gst_play_config_get_seek_accurate (const GstStructure * config);
-- 
2.43.0

